### PR TITLE
#9 GET api/users built and tested with 404 and 200 capabilities

### PR DESCRIPTION
**Pull request by HadrianDeveloper**

### CODE REVIEW

File: controllers/users-controller.js [hunk 1]
```diff
@@ -0,0 +1,9 @@
+const {selectAllUsers} = require("../models/users-model.js");
+const {} = require("../utils/utils.js")
+
+exports.getAllUsers = (req, res, next) => {
```
Comment (JVal-NC) [inline controllers/users-controller.js:4]: catch block required for potential 500 errors

File: __tests__/app.test.js [hunk 2]
```diff
@@ -370,4 +371,24 @@ describe('PATCH /api/articles/:article_id', () => {
         })
     });
 
-});
\ No newline at end of file
+});
+
+describe.only('GET /api/users', () => {
```
Comment (JVal-NC) [inline __tests__/app.test.js:376]: get rid of .only on tests when submitting for pull request

**Review by JVal-NC — COMMENTED** [review]
Minor comments, please include these before subsequent PR.

**Comment by HadrianDeveloper** [comment]
Thanks JVal - this is done. But what is the difference between the catch blocks? 

- .catch(next)
- .catch((err) => next(err))

File: controllers/users-controller.js [hunk 3]
```diff
@@ -0,0 +1,10 @@
+const {selectAllUsers} = require("../models/users-model.js");
+const {} = require("../utils/utils.js")
+
+exports.getAllUsers = (req, res, next) => {
+    selectAllUsers()
+    .then((usersArray) => {
+        res.status(200).send({ allUsers: usersArray })
```
Comment (callumu21) [inline controllers/users-controller.js:7]: This should be a key of just 'users'

File: controllers/users-controller.js [hunk 4]
```diff
@@ -0,0 +1,10 @@
+const {selectAllUsers} = require("../models/users-model.js");
+const {} = require("../utils/utils.js")
+
+exports.getAllUsers = (req, res, next) => {
+    selectAllUsers()
+    .then((usersArray) => {
+        res.status(200).send({ allUsers: usersArray })
+    })
+    .catch(err)
```
Comment (callumu21) [inline controllers/users-controller.js:9]: This catch block needs some code to handle the error. Either .catch(err) {next(err)} or just catch(next).

Both will do the same thing; it's just that the second is refactored. Whatever function you put in the catch brackets will automatically be invoked with error by default, but let me know if you have any questions 👍

**Review by callumu21 — APPROVED** [review]
Just a minor change needed to the catch block. If you can make those changes, you should be good to merge without another PR 👍

**Pull request merged**